dandi/files/zarr.py

```diff
@@ -209,32 +244,43 @@ def get_validation_errors(
         import zarr
 
         errors: list[ValidationResult] = []
+        origin: Origin = Origin(
+            type=OriginType.INTERNAL,
+            validator=Validator.zarr,
+            validator_version=zarr.__version__,
+            standard=Standard.ZARR,
+        )
+
         try:
             data = zarr.open(str(self.filepath))
-        except Exception:
+        except Exception as e:
             if devel_debug:
                 raise
             errors.append(
                 ValidationResult(
-                    origin=ValidationOrigin(
-                        name="zarr",
-                        version=zarr.version.version,
-                    ),
+                    origin=origin,
                     severity=Severity.ERROR,
                     id="zarr.cannot_open",
                     scope=Scope.FILE,
+                    origin_result=e,
                     path=self.filepath,
                     message="Error opening file.",
                 )
             )
             data = None
+
+        origin = Origin(
+            type=OriginType.VALIDATION,
+            validator=Validator.dandi_zarr,
+            validator_version=__version__,
+        )
```

Comment:
here we seems to overwrite prior origin which was like `origin_internal_zarr` and this one `origin_dandi_zarr` or alike.  So might better give this somewhat more descriptive name and avoid overwrite so we do not by mistake assign the wrong one?

Reply:
Rename the first `origin` to `origin_internal_zarr`. The second `origin` is eliminated entirely because I think the subsequent `ValidationResult` objects are labeled incorrectly. Please see https://github.com/dandi/dandi-cli/pull/1514/commits/c09cab4e82947b94da391c757c012dd67e3ee759 for details.